test/e2e/suite_test.go

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2ESuite")
+}
+
+var centralClusterClient client.Client
+var centralClusterKubeconfigData []byte
+var centralClusterNamespace = "e2e-org"
+var centralClusterNamespaceEnvVar = "TEST_E2E_NAMESPACE"
+
+const (
+	timeout  = time.Second * 60
+	interval = time.Second
+)
+
+var _ = BeforeSuite(func() {
```

Comment:
I think most of this is covered in
https://github.com/cloudoperators/greenhouse/blob/main/pkg/test/env.go#L98
Which we should uniformely use to boilerplate setup our test-suites.
env-test will also automatically populate the `config` and the `client` it's initalized with when `USE_EXISTING_CLUSTER=true`

So we call `test.TestBeforeSuite` as in https://github.com/cloudoperators/greenhouse/blob/a0a3d4f873f34c0adda7e0d9b009396f69a96f07/test/e2e/suite_test.go#L33

Reply:
outdated, resolving